iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -555,7 +555,9 @@ private async Task HandleIncomingMessagesAsync()
                 Logging.Enter(this, "Process C2D message via callback", nameof(HandleIncomingMessagesAsync));
 
             Message message = ProcessMessage(true);
-            await (_deviceMessageReceivedListener?.Invoke(message) ?? TaskHelpers.CompletedTask).ConfigureAwait(false);
+            _deviceMessageReceivedListener?.Invoke(message);
```

Comment:
If we can await it, I suppose it returns a task. In which case, if we want to signal to others we don't mean to wait on it, we should:

1. Comment that we intentionally do not await it
2. Receive the task with a discard `_ = _deviceMessageReceivedListener?.Invoke(message);` to signal we don't care about the return value

Reply:
sure thing, will add